Tests/UnitTests/MessageBirdUnitTests/Resources/ClientVerifyTest.cs

```diff
@@ -0,0 +1,151 @@
+﻿using MessageBird.Net;
+using MessageBird.Objects;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using WireMock.Server;
+
+namespace MessageBirdTests.Resources
+{
+    /**
+     * Using this class to be able to change the base url of the request.
+     */ 
+    internal class VerifyWithUrl : MessageBird.Resources.Resource
+    {
+        public VerifyWithUrl(Verify verify)
+            : base("verify", verify)
+        {
+        }
+
+        private string VerifyBaseUrl;
+
+        private string Token
+        {
+            get
+            {
+                return (Object != null) ? ((Verify)Object).Token : null;
+            }
+        }
+
+        private bool HasToken
+        {
+            get
+            {
+                return !string.IsNullOrEmpty(Token);
+            }
+        }
+
+        public override string QueryString
+        {
+            get
+            {
+                return HasToken ? "token=" + System.Uri.EscapeDataString(Token) : string.Empty;
+            }
+        }
+
+        public void SetBaseUrl(string baseUrl)
+        {
+            VerifyBaseUrl = baseUrl;
+        }
+
+        public override string BaseUrl => VerifyBaseUrl;
+    }
+
+    [TestClass]
+    public class RestClientVerifyTest
+    {
+        private const string ENDPOINT = "http://localhost:";
+        private const string ACCESS_KEY = "SOME_ACCESS_KEY";
+        private const long RECIPIENT = 31612345678;
+        private const string TEMPLATE = "test template";
+        private const string ORIGINATOR = "test ori";
+        private const DataEncoding DATA_ENCODING = DataEncoding.Auto;
+        private const string REFERENCE = "testref";
+        private const int TIMEOUT = 1234;
+        private const int TOKEN_LENGTH = 15;
+        private const Voice VOICE = Voice.Female;
+        private const Language LANGUAGE = Language.BrazilianPortugese;
+
+
+        private static string CREATE_VERIFY_RESPONSE = "{" +
+                "\"id\": \"2cec4f419b7a4cf7b546a41c630a20b0\"," +
+                "\"href\": \"some_url\"," +
+                "\"recipient\": 31627343907," +
+                "\"reference\": null," +
+                "\"messages\": {" +
+                "    \"href\": \"some_url\"" +
+                "}," +
+                "\"status\": \"sent\"," +
+                "\"createdDatetime\": \"2019-07-04T08:23:37+00:00\"," +
+                "\"validUntilDatetime\": \"2019-07-04T08:24:07+00:00\"" +
+                "}";
+
+        private static string EXPECTED_REQUEST = "{" +
```

Comment:
I would make this a const string instead of static, unless you are expecting it to change at some point